### PR TITLE
[Modules] Updated API reference (Microsoft.Authorization/roleDefinition) to the newest one. Issue #1998

### PR DESCRIPTION
**Pull request by cjnova**
# Description

>Thank you for your contribution !

> Please include a summary of the change and which issue is fixed.
Updated API version referenced, related to issue #1998 
> Please also include the context.
Microsoft.Authorization/roleDefinitions
> List any dependencies that are required for this change.
None

| Pipeline |
| - |
[Authorization: RoleDefinitions](https://github.com/cjnova/ResourceModules/actions/workflows/ms.authorization.roledefinitions.yml)
| |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Comment by cjnova** [comment]
Fix for Issue #1998

File: modules/README.md [hunk 1]
```diff
@@ -4,110 +4,110 @@ In this section you can find useful information regarding the Modules that are c
 
 | Name | Provider namespace | Resource Type |
 | - | - | - |
-| [Azure Active Directory Domain Services](https://github.com/Azure/ResourceModules/tree/main/modules/Microsoft.AAD/DomainServices) | `MS.AAD` | [DomainServices](https://github.com/Azure/ResourceModules/tree/main/modules/Microsoft.AAD/DomainServices) |
```
Comment (ahmadabdalla) [inline modules/README.md:7]: Please undo changes to this file

File: README.md [hunk 2]
```diff
@@ -30,113 +30,113 @@ The CI environment supports both ARM and Bicep and can be leveraged using GitHub
 
 | Name | Status |
 | - | - |
-| [Action Groups](https://github.com/Azure/ResourceModules/tree/main/modules/Microsoft.Insights/actionGroups) | [!['Insights: ActionGroups'](https://github.com/Azure/ResourceModules/workflows/Insights:%20ActionGroups/badge.svg)](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.actiongroups.yml) |
```
Comment (ahmadabdalla) [inline README.md:33]: Please undo changes to this file

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]
Thank you for your contribution. Just minor changes to files that need to be reverted

**Review by ahmadabdalla — APPROVED** [review]
Thanks :) approved

**Pull request merged**